### PR TITLE
Fix/transaction crashes 240

### PR DESCRIPTION
**Pull request by kcw-grunt**
# Describe the issue
Many users where experience a runtime crash during sync.  It turns on the the syncing process is not at all performant and there was a very likely scenario where the user addresses would overflow the mem space during sync

This fix resolves and checks the data first

### CODE REVIEW

**Comment by kcw-grunt** [comment]
@hassanashraf92  will you convert this to a regular PR when you are comfortable?  
 Please invite me and @josikie to review.

File: litewallet.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist
Comment (kcw-grunt) [inline litewallet.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:1]: one of the dev leaks I have ignored so long @hassanashraf92

File: litewallet/BartyCrouch.swift [hunk 1]
```diff
@@ -24,8 +24,7 @@ enum BartyCrouch {
 		case turkey = "tr"
 	}
 
-	static func translate(key: String, translations: [SupportedLanguage: String], comment _: String? = nil) -> String
-	{
+	static func translate(key: String, translations: [SupportedLanguage: String], comment _: String? = nil) -> String {
 		let typeName = String(describing: BartyCrouch.self)
 		let methodName = #function
```
Comment (josikie) [inline litewallet/BartyCrouch.swift:29]: What is #function?

File: litewallet/Environment.swift [hunk 2]
```diff
@@ -126,20 +126,18 @@ struct E {
...
+		return UIScreen.main.bounds.size.height == 480.0
 	}
 
 	static var isIPhone5: Bool {
 		return (UIScreen.main.bounds.size.height == 568.0) && (E.is32Bit)
 	}
 
 	static var isIPhoneX: Bool {
-		return (UIScreen.main.bounds.size.height == 812.0)
+		return UIScreen.main.bounds.size.height == 812.0
```
Comment (josikie) [inline litewallet/Environment.swift:137]: nice

File: litewallet/Environment.swift [hunk 3]
```diff
@@ -126,20 +126,18 @@ struct E {
...
 	}
 
-	static let is32Bit: Bool = {
-		MemoryLayout<Int>.size == MemoryLayout<UInt32>.size
-	}()
+	static let is32Bit: Bool = MemoryLayout<Int>.size == MemoryLayout<UInt32>.size
```
Comment (josikie) [inline litewallet/Environment.swift:140]: Is it allowed to compare Int data type to UInt32 data type in Swift?

File: litewallet/MainViewController.swift [hunk 4]
```diff
@@ -141,23 +141,21 @@ class MainViewController: UIViewController, Subscriber, LoginViewControllerDeleg
...
+		}
 
-		NotificationCenter.default.addObserver(forName: UIScene.willDeactivateNotification, object: nil, queue: nil)
-			{ _ in
-				if !self.isLoginRequired, !self.store.state.isPromptingBiometrics {
-					self.blurView.alpha = 1.0
-					self.view.addSubview(self.blurView)
-					self.blurView.constrain(toSuperviewEdges: nil)
-				}
+		NotificationCenter.default.addObserver(forName: UIScene.willDeactivateNotification, object: nil, queue: nil) { _ in
```
Comment (josikie) [inline litewallet/MainViewController.swift:152]: what is ` _ in`? I saw it many times the `_` doesn't have comma, only space

File: litewallet/src/Extensions/CGContext+Additions.swift [hunk 5]
```diff
@@ -4,8 +4,8 @@ extension CGContext {
 	func addLineThrough(_ points: [(CGFloat, CGFloat)]) {
 		guard let first = points.first else { return }
 		move(to: CGPoint(x: first.0, y: first.1))
-		points.dropFirst().forEach {
-			addLine(to: CGPoint(x: $0.0, y: $0.1))
+		for point in points.dropFirst() {
```
Comment (josikie) [inline litewallet/src/Extensions/CGContext+Additions.swift:7]: Nice

**Review by josikie — APPROVED** [review]
Thank you @kcw-grunt and @hassanashraf92

**Pull request merged**